Ghost_in_the_Shell/client_folder_react_app/ethisim/src/components/EditorComponents/FlowDiagram.js

```diff
@@ -0,0 +1,262 @@
+    return (
+        <>
+            <Handle type="target" position="top" />
+            {data.label}
+            <Handle type="source" position="bottom" />
+        </>
+    );
+};
+
+const genericNode = ({ data }) => {
+    return (
+        <>
+            <Handle type="target" position="top" />
+            {data.label}
+            <Handle type="source" position="bottom" />
+        </>
+    );
+};
+
+const conversationNode = ({ data }) => {
+    return (
+        <>
+            <Handle type="target" position="top" />
+            {data.label}
+            <Handle type="source" position="bottom" />
+        </>
+    );
+};
+
+function initializeElements(componentData) {
+    if (componentData.type === 'Action') {
+        return {
+            id: componentData.id,
+            type: 'actionNode',
+            data: { label: componentData.title },
+            style: {
+                border: '3px solid green',
+                borderRadius: '5%',
+                padding: 10,
+            },
+            position: componentData.position,
+        };
+    } else if (componentData.type === 'Generic') {
+        return {
+            id: componentData.id,
+            type: 'genericNode',
+            data: { label: componentData.title },
+            style: { border: '3px solid red', borderRadius: '5%', padding: 10 },
+            position: componentData.position,
+        };
+    } else if (componentData.type === 'Reflection') {
+        return {
+            id: componentData.id,
+            type: 'reflectionNode',
+            data: { label: componentData.title },
+            style: {
+                border: '3px solid purple',
+                borderRadius: '5%',
+                padding: 10,
+            },
+            position: componentData.position,
+        };
+    } else if (componentData.type === 'Conversation') {
+        return {
+            id: componentData.id,
+            type: 'conversationNode',
+            data: { label: componentData.title },
+            style: {
+                border: '3px solid blue',
+                borderRadius: '5%',
+                padding: 10,
+            },
+            position: componentData.position,
+        };
+    }
+}
+export default function FlowDiagram(props) {
+    const initialElements = mockComponentsFlowChart.components.map(
+        (componentData) => {
+            return initializeElements(componentData);
+        }
+    );
+
+    const [elements, setElements] = useState(initialElements);
+    const [isRemoveButtonDisabled, setIsRemoveButtonDisabled] = useState(true);
+    const [currentEdgeSelected, setCurrentEdgeSelected] = useState();
+
+    const classes = useStyles();
+
+    const nodeTypes = {
+        actionNode: actionNode,
+        reflectionNode: reflectionNode,
+        genericNode: genericNode,
+        conversationNode: conversationNode,
+    };
+
+    const graphStyles = { width: '100%', height: '500px', border: 'solid' };
```

Comment:
Can this hard coded px count be changed to something more responsive?

Reply:
Changed for flow diagram to be responsive to browser height (Set height to 100%, sent container div to height: 90vh to avoid overflow-y)